changes in product and category.

samisaeed626 opened the pull request:
added new fields, design improvements related changes

samisaeed626 committed:
changes in product and category.

added new fields, design improvements related changes

abdullahzulfiqar653 commented on src/products/models.py:
```diff
@@ -32,16 +33,21 @@ class Product(models.Model):
     )
     category = models.ForeignKey(
-        Category, related_name="category", on_delete=models.CASCADE, db_column='category')
+        Category, related_name="category", on_delete=models.CASCADE, db_column='category',null=True)
```
add space after comma

abdullahzulfiqar653 commented on src/products/models.py:
```diff
@@ -32,16 +33,21 @@ class Product(models.Model):
+        Category, related_name="category", on_delete=models.CASCADE, db_column='category',null=True)
 
-    price = models.FloatField(default=0)
+    # price = models.FloatField(default=0)
```
remove commented field

abdullahzulfiqar653 commented on src/products/views.py:
```diff
@@ -154,17 +154,29 @@ def ProductsAddView(request):
-
+        category_id = data.get('category', '')
+        expire_date = data.get('expire_date', '')
+        if category_id and category_id.isdigit():#if category not enter from frontend
```
no need to check if category id is digit or not

samisaeed626 committed:
all sale update
.add new fields
.update and improve design
.add LocalStorage for hold system

samisaeed626 committed:
Updates and improvement
.pdf update and improve
.project name change
.sele view update
.sale print improve and update
.sale_details  table  reverce

abdullahzulfiqar653 commented on src/products/urls.py:
```diff
@@ -27,4 +27,5 @@
          views.ProductsDeleteView, name='products_delete'),
     # Get products AJAX
     path("get", views.GetProductsAJAXView, name="get_products"),
+    
```
remove line 30 an extra line it is

abdullahzulfiqar653 commented on src/products/views.py:
```diff
@@ -229,10 +229,13 @@ def ProductsUpdateView(request, product_id):
-            if category_id and category_id.isdigit():
+            expire_date = data.get('expire_date', '')
+            if category_id == '':
                 category = Category.objects.get(id=int(category_id))
```
check this condition

abdullahzulfiqar653 commented on src/products/views.py:
```diff
@@ -229,10 +229,13 @@ def ProductsUpdateView(request, product_id):
             else:
                 category = None 
+            if expire_date == '': #if expire date not enter from frontend
+               expire_date=None    
```
put space between like this expire_date = None

abdullahzulfiqar653 commented on src/products/views.py:
```diff
@@ -243,10 +246,10 @@ def ProductsUpdateView(request, product_id):
-                "expire_date": data['expire_date'],
+                "expire_date": expire_date,
             }
             print(attributes)
```
remove print

abdullahzulfiqar653 commented on src/sales/models.py:
```diff
@@ -7,13 +7,19 @@
     date_added = models.DateTimeField(default=django.utils.timezone.now)
     customer = models.ForeignKey(
-        Customer, models.DO_NOTHING, db_column='customer')
+    Customer, models.DO_NOTHING, db_column='customer',null=True)
```
add a space after comma

abdullahzulfiqar653 commented on src/sales/urls.py:
```diff
@@ -1,10 +1,11 @@
 app_name = "sales"
 urlpatterns = [
     # List sales
+     path('get_customers_ajax', views.get_customers_ajax, name='get_customers_ajax'),
```
remove one space from the start of first path.

abdullahzulfiqar653 commented on src/sales/views.py:
```diff
@@ -27,23 +33,35 @@ def SalesListView(request):
+            if customer == 'walking_customer':
+                customer = None
+            else:
+                customer = Customer.objects.get(id=int(customer)) 
```
if customer id is string then try passing it without int parsing

abdullahzulfiqar653 commented on src/sales/views.py:
```diff
@@ -27,23 +33,35 @@ def SalesListView(request):
+                customer = None
+            else:
+                customer = Customer.objects.get(id=int(customer)) 
+            if dicount == '':
```
make this condition better

abdullahzulfiqar653 commented on src/static/css/sb-admin-2.min.css:
```diff
@@ -12614,3 +12614,7 @@ body.sidebar-toggled footer.sticky-footer {
 
+.select2-container--default .select2-results__option--highlighted.select2-results__option--selectable {
+    background-color: #35373b;
+    color: white;
```
add enpty line at the end of the file

abdullahzulfiqar653 commented on src/templates/sales/sales.html:
```diff
@@ -46,12 +46,12 @@ <h6 class="m-0 font-weight-bold text-primary">Customers</h6>
                     </thead>
                     <tbody>
-                        {% for s in sales %}
+                        {% for s in sales|slice:"::-1" %}
```
use a good variable in instead of s like sale in sales

abdullahzulfiqar653 commented on src/templates/sales/sales_receipt_pdf.html:
```diff
@@ -55,13 +55,18 @@
-            <p><b>Customer:</b> {{sale.customer.get_full_name}}</p>
+            <p><b>Customer:</b> {% if sale.customer.get_full_name is None %}
+                walking Customer
+                {%else%}
```
put space after % and before % like {% else %}

abdullahzulfiqar653 commented on src/templates/sales/sales_receipt_pdf.html:
```diff
@@ -55,13 +55,18 @@
+                walking Customer
+                {%else%}
+            {{sale.customer.get_full_name}}
+            {%endif%}
```
put space after % and before % like {% endif %}

samisaeed626 committed:
update all resolve conversation
